ldfparser/parser.py

```diff
@@ -104,128 +156,226 @@ def parse_real_or_integer(self, i:str):
+
+	def header_channel(self, tree):
+		return ("channel_name", tree[0])
+
+	def nodes(self, tree):
+		return ("nodes", {'master': tree[0], 'slaves': tree[1]})
+
+	def nodes_master(self, tree):
+		return {"name": tree[0], "timebase": tree[1] * 0.001, "jitter": tree[2] * 0.001}
+
+	def nodes_slaves(self, tree):
+		return tree
+
+	def node_compositions(self, tree):
+		return ("node_compositions", tree[0:])
+
+	def node_compositions_configuration(self, tree):
+		return {"name": tree[0], "compositions": tree[1]}
+
+	def node_compositions_composite(self, tree):
+		return {"name": tree[0], "nodes": tree[1:]}
+
+	def signals(self, tree):
+		return ("signals", tree)
+
+	def signal_definition(self, tree):
+		return {"name": tree[0], "width": int(tree[1]), "init_value": tree[2], "publisher": tree[3], "subscribers": tree[4:]}
+
+	def signal_default_value(self, tree):
+		return tree[0]
+
+	def signal_default_value_single(self, tree):
+		return int(tree[0])
+
+	def signal_default_value_array(self, tree):
+		return tree[0]
+
+	def frames(self, tree):
+		return ("frames", tree)
+
+	def frame_definition(self, tree):
+		return {"name": tree[0], "frame_id": int(tree[1]), "publisher": tree[2], "length": tree[3] if len(tree) > 4 else None, "signals": tree[4] if len(tree) > 4 else tree[3]}
+
+	def frame_signals(self, tree):
+		return tree[0:]
+
+	def frame_signal(self, tree):
+		return {"signal": tree[0], "offset": int(tree[1])}
+
+	def sporadic_frames(self, tree):
+		return ("sporadic_frames", tree[0:])
+
+	def sporadic_frame_definition(self, tree):
+		return {"name": tree[0], "frames": tree[1:]}
+
+	def event_triggered_frames(self, tree):
+		return ("event_triggered_frames", tree[0:])
+
+	def event_triggered_frame_definition(self, tree):
+		return {"name": tree[0]}
+
+	def event_triggered_frame_definition_frames(self, tree):
+		return tree[0:]
+
+	def node_attributes(self, tree):
+		return ("node_attributes", tree[0:])
+
+	def node_definition(self, tree):
+		node = {"name": tree[0]}
+		for k in tree[1:]:
+			node[k[0]] = k[1]
+		return node
+
+	def node_definition_protocol(self, tree):
+		return ("lin_protocol", tree[0])
+
+	def node_definition_configured_nad(self, tree):
+		return ("configured_nad", tree[0])
+
+	def node_definition_initial_nad(self, tree):
+		return ("initial_nad", tree[0])
+
+	def node_definition_product_id(self, tree):
+		return ("product_id", {"supplier_id": tree[0], "function_id": tree[1], "variant": tree[2] if len(tree) > 2 else None})
+
+	def node_definition_response_error(self, tree):
+		return ("response_error", tree[0])
+
+	def node_definition_fault_state_signals(self, tree):
+		return ("fault_state_signals", tree[0:])
+
+	def node_definition_p2_min(self, tree):
+		return ("P2_min", tree[0] * 0.001)
+
+	def node_definition_st_min(self, tree):
+		return ("ST_min", tree[0] * 0.001)
+
+	def node_definition_n_as_timeout(self, tree):
+		return ("N_As_timeout", tree[0] * 0.001)
+
+	def node_definition_n_cr_min(self, tree):
```

Comment:
Typo, should be the same as in the LIN spec and in the lark file (already fixed):

```python
def node_definition_n_cr_timeout(self, tree):
```